### PR TITLE
Singleton http client

### PR DESCRIPTION
**Pull request by UberMouse**
Instead of creating a new HttpClient every time an exception is sent, share a singleton instance as per #321. Make it a static Property with a setter so it can be supplied by the user as per #322.

---
This change is [Reviewable](https://reviewable.io/reviews/mindscapehq/raygun4net/327)

### CODE REVIEW

File: Mindscape.Raygun4Net.WebApi/packages.config [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.AspNet.WebApi.Client" version="5.1.1" targetFramework="net45" />
-  <package id="Microsoft.AspNet.WebApi.Core" version="5.1.1" targetFramework="net45" />
-  <package id="Newtonsoft.Json" version="4.5.11" targetFramework="net45" />
+  <package id="Microsoft.AspNet.WebApi.Client" version="5.2.3" targetFramework="net45" />
+  <package id="Microsoft.AspNet.WebApi.Core" version="5.2.3" targetFramework="net45" />
+  <package id="Newtonsoft.Json" version="6.0.4" targetFramework="net45" />
```
Comment (j5alive) [inline Mindscape.Raygun4Net.WebApi/packages.config]: Did these packages need to be updated? Do we even use Newtonsoft.Json in this project?

File: Mindscape.Raygun4Net.WinRT/RaygunClient.cs [hunk 2]
```diff
@@ -336,9 +341,9 @@ protected IEnumerable<Exception> StripWrapperExceptions(Exception exception)
     private async Task PostMessageAsync(HttpClient client, HttpContent httpContent, Uri uri)
 #pragma warning restore 1998
     {
+      // Unsure why this is structured like this, don't want to change incase it's for a valid reason
```
Comment (j5alive) [inline Mindscape.Raygun4Net.WinRT/RaygunClient.cs]: I would say the statements can be joined and then you can remove this comment.

File: Mindscape.Raygun4Net.WebApi/packages.config (hunk 1 above)
Comment (UberMouse) [inline Mindscape.Raygun4Net.WebApi/packages.config]: Had to update WebApi.Core to work with the new WebApi version (Raygun4Net.WebApi didn't work in my brand new WebApi project) and I believe that WebApi.Client depends on NewtonSoft.Json >= 6.0.4

**Comment by grugcrood82** [comment]
Resolved by #409

**Comment by BarsikV** [comment]
This was first resolved by #409 , but then the PR was reverted by #410 . Please fix the issue with HttpClient and make it Singleton, static or configurable. It's harming our services.

**Pull request closed without merging**